ext/xmlrpc/libxmlrpc/encodings.c

```diff
@@ -67,26 +67,24 @@ static char* convert(const char* src, int src_len, int *new_len, const char* fro
          size_t st;
          outbuf = (char*)emalloc(outlen + 1);
 
-         if(outbuf) {
-            out_ptr = (char*)outbuf;
-            while(inlenleft) {
-               st = iconv(ic, (char**)&src, &inlenleft, &out_ptr, &outlenleft);
-               if(st == -1) {
-                  if(errno == E2BIG) {
-                     int diff = out_ptr - outbuf;
-                     outlen += inlenleft;
-                     outlenleft += inlenleft;
-                     outbuf = (char*)erealloc(outbuf, outlen + 1);
-                     if(!outbuf) {
-                        break;
-                     }
-                     out_ptr = outbuf + diff;
-                  }
-                  else {
-                     efree(outbuf);
-                     outbuf = 0;
+         out_ptr = (char*)outbuf;
```

Comment:
Is the cast needed ?

Reply:
Since [`outbuf` is declared as `char *`](https://github.com/staabm/php-src/blob/7a99f0e9fccab904b27403b3f3908b03162da6d5/ext/xmlrpc/libxmlrpc/encodings.c#L53), no. It doesn't hurt, though.